test/utils.go

```diff
@@ -0,0 +1,204 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+	"golang.org/x/oauth2/google"
+	cloudresourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+	alpha "google.golang.org/api/compute/v0.alpha"
+	beta "google.golang.org/api/compute/v0.beta"
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+	"k8s.io/klog/v2"
+	boskosclient "sigs.k8s.io/boskos/client"
+	"sigs.k8s.io/boskos/common"
+)
+
+var boskos, _ = boskosclient.NewClient(os.Getenv("JOB_NAME"), "http://boskos", "", "")
+
+// getBoskosProject retries acquiring a boskos project until success or timeout.
+func getBoskosProject(resourceType string) *common.Resource {
+	timer := time.NewTimer(30 * time.Minute)
+	defer timer.Stop()
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-timer.C:
+			klog.Fatalf("timed out trying to acquire boskos project")
+		case <-ticker.C:
+			p, err := boskos.Acquire(resourceType, "free", "busy")
+			switch {
+			case err != nil:
+				klog.Warningf("boskos failed to acquire project: %w", err)
+			case p == nil:
+				klog.Warningf("boskos does not have a free %s at the moment", resourceType)
+			default:
+				return p
+			}
+		}
+	}
+}
+
+func setupProwConfig(resourceType string) (string, string) {
+	// Try to get a Boskos project
+	klog.V(4).Info("Running in PROW")
+	klog.V(4).Info("Fetching a Boskos loaned project")
+
+	p := getBoskosProject(resourceType)
+	project := p.Name
+
+	go func(c *boskosclient.Client, _ string) {
+		for range time.Tick(time.Minute * 5) {
+			if err := c.UpdateOne(p.Name, "busy", nil); err != nil {
+				klog.Warningf("[Boskos] Update %s failed with %v", p.Name, err)
+			}
+		}
+	}(boskos, p.Name)
+
+	klog.V(4).Info("Fetching the default compute service account")
+
+	c, err := google.DefaultClient(context.Background(), cloudresourcemanager.CloudPlatformScope)
+	if err != nil {
+		klog.Fatalf("Failed to get Google Default Client: %w", err)
+	}
+
+	cloudresourcemanagerService, err := cloudresourcemanager.NewService(context.Background(), option.WithHTTPClient(c))
+	if err != nil {
+		klog.Fatalf("Failed to create new cloudresourcemanager: %w", err)
+	}
+
+	resp, err := cloudresourcemanagerService.Projects.Get(project).Do()
+	if err != nil {
+		klog.Fatalf("Failed to get project %v from Cloud Resource Manager: %w", project, err)
+	}
+
+	// Default Compute Engine service account
+	// [PROJECT_NUMBER]-compute@developer.gserviceaccount.com
+	serviceAccount := fmt.Sprintf("%v-compute@developer.gserviceaccount.com", resp.ProjectNumber)
+	klog.Infof("Using project %v and service account %v", project, serviceAccount)
```

Comment:
Use the most specific formatting string directive for a variable type:

%s for string etc. Don't default to %v